File: June 2022/June_10.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int ans = 0, i = 0, j = 0, n = s.length();
        int[] freq = new int[256];
        while(j < n){
            char ch = s.charAt(j);
            while(freq[ch] > 0)
                freq[s.charAt(i++)]--;
            freq[ch]++;
            ans = Math.max(ans, j - i + 1);
            j++;
        }
        return ans;
    }
}

File: June 2022/June_10.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> freq(256, 0);
        int j = 0, len = 0;
        for(int i = 0; i < s.length(); i++){
            int ch = s[i];
            while(freq[ch] > 0)
                freq[s[j++]]--;
            freq[ch]++;
            len = max(len, i - j + 1);
        }
        return len;
    }
};
